Upgrade to numpy 2.0

nate-kean opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
- Sets RAiDER's minimum version requirement for numpy to 2.0.
- Migrates away from functions deprecated in numpy 2.0.
- Adds the numpy 2.0 linting ruleset to ruff.

Resolves https://github.com/dbekaert/RAiDER/issues/691.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
Issue was prompted by RAiDER's conda feedstock: https://github.com/conda-forge/raider-feedstock/pull/19

## How Has This Been Tested?
<!--- Please run the following command on your PR to ensure code formatting consistency: -->
<!--- autopep8 <changed files/folders> --recursive --in-place --ignore=E5 -->
- Tests continue to pass
- No tests added
- No new warnings


## Type of change
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have added an explanation of what your changes do and why you'd like us to include them.
- [ ] I have written new tests for your core changes, as applicable.
- [x] I have successfully ran tests with your changes locally.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

nate-kean committed:
Set minimum numpy version to 2.0

nate-kean committed:
Add ruff ruleset for numpy 2.0 migration

nate-kean committed:
Migrate np.trapz to np.trapezoid

nate-kean committed:
Update CHANGELOG.md for #746

jlmaurer commented:
@garlic-os one thing I think we might should do here is to maintain backwards compatibility with Numpy < 2. This is in particular for people who are already running raider to prevent them having to upgrade numpy. The easiest way to do this is probably to define our own trapz function that uses a try/except to load the correct numpy function, then the rest of RAiDER can call that custom function. I know it's clunky but since we're running enterprise now we should probably take that approach.

nate-kean committed:
Support both numpy v1 and v2

- environment.yml: Remove requirement for numpy to be at least v2
- Add polyfill for np.trapezoid for numpy v1

nate-kean commented:
How is this?
